modules/cf-core/src/message-handling/handle-protocol-message.ts

```diff
@@ -280,15 +280,20 @@ async function getQueueNamesListByProtocolName(
   params: ProtocolParameters,
   requestHandler: RequestHandler
 ): Promise<string[]> {
-  const { publicIdentifier, networkContext, store } = requestHandler;
+  const {
+    networkContext,
+    provider,
+    publicIdentifier,
+    store
+  } = requestHandler;
 
   async function multisigAddressFor(xpubs: string[]) {
     const allowGenerated = protocol === Protocol.Setup || protocol === Protocol.InstallVirtualApp;
     return await store.getMultisigAddressWithCounterparty(
       xpubs,
       networkContext.ProxyFactory,
       networkContext.MinimumViableMultisig,
-      allowGenerated,
+      allowGenerated ? provider : undefined
```

Comment:
technically, in the install virtual app protocol, it should only `allowGenerated` for the channel between the initiator and the responder, but not be allowed for the initiator and intermediary or the responder and intermediary